### DISCUSSION TITLE
code generation capability?

### DISCUSSION THREAD

**Discussion by dyxribo** [BowlerHatLLC/vscode-as3mxml#738] [Q&A] [answered]
hey all ! 

i have a simple question-- forgive me if it's already been asked, but i couldn't find any similar posts with the terms i searched.

does this extension have a code generation capability that i'm missing? like generating a function from context, such as a function call or from an addEventListener statement? it was a great feature i had available to me in flashdevelop and it saved a bit of time and keystrokes. 

if there's a separate extension for that already, that would be fine too!

**Comment by joshtynjala** [comment, marked as answer]
See [vscode-as3mxml: Code Generation](https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/Code-Generation).

These are exposed as "Code Actions" in VSCode. You should see a little light bulb appear when you click on a symbol name in the editor that is red squiggly underlined with an error about a missing variable/method.

**Reply by dyxribo** [reply]
wow, that was embarrassingly in your face, i've been trying to use the "source action" and "refactor" options on the right click context menu 😅 this is perfect. thank you, and thank you for helping to keep the as3 community alive!!
